### PR TITLE
ACMS 1623

### PR DESCRIPTION
**Pull request by vipin-mittal-acquia**
pipeline-fail-security-patch-available
drush-security-check
**Motivation**
Include pm:security command in the CI pipeline to know the security patches released in core/contrib

**Proposed changes**
Added CI job with command `./vendor/bin/drush pm:security`

**Alternatives considered**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: .github/workflows/drush-security-check.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Drush Security Check
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  check:
+    name: PHP ${{ matrix.php-versions }}
```
Comment (rajeshreeputra) [inline .github/workflows/drush-security-check.yml:8]: Can we change name:
`name: Drush Security Check for PHP ${{ matrix.php-versions }}`
@vipin-mittal-acquia @vishalkhode1

Comment (vipin-mittal-acquia) [reply .github/workflows/drush-security-check.yml:8]: @rajeshreeputra : Please refer to the attached [screenshot](https://user-images.githubusercontent.com/109503136/211245454-26fd672b-bb25-4158-866f-44e73921853c.png) to see what you mean by the changes.

File: .github/workflows/drush-security-check.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        php-versions: ['8.1', '8.2']
```
Comment (vishalkhode1) [inline .github/workflows/drush-security-check.yml:12]: @vipin-mittal-acquia I think there's no need to check this on separate PHP version.

File: .github/workflows/drush-security-check.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          php-version: ${{ matrix.php-versions }}
+
+      - name: Install Drupal core dependencies and modules
+        run: |
+          composer --no-interaction --no-progress --prefer-dist --optimize-autoloader install
+
+      - name: Run security check
+        run: |
+          ./vendor/bin/drush pm:security
```
Comment (vishalkhode1) [inline .github/workflows/drush-security-check.yml:28]: @vipin-mittal-acquia Does CI fails if there's any security issue ? How do would know, if we are using insecure module ?

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: .github/workflows/drush-security-check.yml (hunk 3 above)
Comment (vipin-mittal-acquia) [reply .github/workflows/drush-security-check.yml:28]: Yes @vishalkhode1 , the pipeline will fail if any security vulnerabilities are discovered in insecure module.

Comment (vishalkhode1) [reply .github/workflows/drush-security-check.yml:28]: Great, if this is returning non-zero exit fail, if there are security issues, then we are good. 😊

Comment (vipin-mittal-acquia) [reply .github/workflows/drush-security-check.yml:28]: @vishalkhode1 : Refer [Screenshot](https://user-images.githubusercontent.com/109503136/211488790-54f3a0c4-28e7-4357-aee4-1f13ed847dde.png), if pipeline fails.

File: .github/workflows/drush-security-check.yml (hunk 1 above)
Comment (vipin-mittal-acquia) [reply .github/workflows/drush-security-check.yml]: @rajeshreeputra: Made changes accordingly

File: .github/workflows/drush-security-check.yml (hunk 2 above)
Comment (vipin-mittal-acquia) [reply .github/workflows/drush-security-check.yml]: @vishalkhode1 : Removed multiple PHP versions check and keep 8.1 default with ubuntu-latest

**Review by vishalkhode1 — APPROVED** [review]

**Pull request merged**